### ISSUE TITLE
Windows 2012 R2 - PS compatibility

### ISSUE THREAD

**Issue by LordDeimos666** [awarre/Optimize-WsusServer#3]
As stated here: https://serverfault.com/questions/746675/how-can-i-use-iis-administration-commandlets-on-windows-server-2012-r2

Your script doens't work on 2012R2 server.

Regards

**Comment by awarre** [comment]
Interesting, I'm surprised this never came up before.

You could try replacing **Get-IISSite** with [Get-Website](https://docs.microsoft.com/en-us/powershell/module/webadministration/Get-Website?view=winserver2012r2-ps). For the purposes of this script they should operate identically.

Please let me know if that works.

I don't have a 2012R2 server to test on, so there may be additional functionality broken beyond just Get-IISSite.

**Comment by LordDeimos666** [comment]
I'll let you know when I have time to do it on my homelab first :)

**Comment by LordDeimos666** [comment]
Long story short. I couldnt' wait (patience over 9000) for it and had some spare time.
Resolved the **Get-IISSite** problem the other cmdlet for IIS installing the nuget package: [https://www.powershellgallery.com/packages/IISAdministration/1.1.0.0](url)

Everything working fine, just one problem I saw, but not really expert on PS scripting side tbh, here i'm running a localized WSUS (italian) and the: `IIS:\Sites\WSUS Administration\ClientWebService` harcoded is not the same as on the italian side, instead it is `IIS:\Sites\Amministrazione di Windows Server Update Services\ClientWebService` maybe it's possible to retrieve the name for it using a cmdlet for the IIS.

Regards
